src/tools/metrics/recorder.rs

```diff
@@ -24,44 +26,68 @@ impl Default for TestMetrics {
 }
 
 impl TestMetrics {
-    pub fn get_val_for(&self, kind: MetricKind, metric: &'static str) -> MetricVal {
-        let key = CompositeKey::new(kind, Key::from_name(metric));
-
-        match &self.0.snapshot().into_hashmap().get(&key).unwrap().2 {
-            DebugValue::Counter(val) => MetricVal::Counter(*val),
-            DebugValue::Gauge(val) => MetricVal::Gauge(val.into_inner()),
-            DebugValue::Histogram(vals) => {
-                MetricVal::Histogram(vals.iter().map(|val| val.into_inner()).collect())
-            }
+    pub fn take_snapshot(&self) -> HashMap<CompositeKey, MetricVal> {
```

Comment:
Perhaps we should make this return a new type that then implements `get_counter`, `get_gauge`, etc...?